### PR TITLE
Record address email owners in ABook imports

### PR DESCRIPTION
**Pull request by leogrim**
@stkem @raymondng

### CODE REVIEW

File: kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterActor.scala [hunk 1]
```diff
@@ -168,6 +180,7 @@ class ContactsUpdater @Inject() (
             abookEntry = db.readWrite(attempts = 2) { implicit s =>
               try {
                 econtactRepo.insertAll(userId, econtactsToAdd.toSeq)
+                s.onTransactionSuccess { recordContactUserIds(econtactsToAdd.map(_.email)) }
```
Comment (stkem) [inline kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterActor.scala:183]: Why not use a change listener on the repo instead of hooking into the transaction? (I just pushed the boilerplate for that, subset of the abook wti hookup)

**Comment by stkem** [comment]
Made one small comment. lgtm otherwise.

File: kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterPluginModule.scala [hunk 2]
```diff
@@ -5,12 +5,11 @@ import com.google.inject.{Provider, Provides}
 import com.keepit.common.actor.ActorInstance
 import akka.actor.ActorSystem
```
Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterPluginModule.scala:7]: @martinraison I removed ABookServiceClientModule from ContactsUpdatedPluginModule and moved it to AbookModule. Two main reasons: 
- Nested modules make a service's dependencies hard to track.
- Potential conflicts if you end up installing two modules that install the same nested module.

As a rule of thumb, you should not have to use "install" in a "Functional Module" (i.e. a module that configures a specific functionality, such as ContactsUpdaterPluginModule). Only a "Configuration module" (i.e. a module that wires a service as a whole) should use "install" on the list of functional modules that it owns. In the latter case, you don't even have to use "install" yourself as the "ConfigurationModule" trait does it using reflection. 

Long story short, if you end up using "install", there might be something fishy. (One example where we do use "install" is CacheModule: we're sure that CachePlugin will never be accessed / injected on their own anywhere else than in CacheModule, so it's ok to encapsulate and install them there.) 

For more about our Guice pattern: 
http://eng.42go.com/guice-in-a-multi-project-play-environment-configuration-of-functional-modules-pattern/

Comment (martinraison) [inline kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterPluginModule.scala:7]: Thanks a lot for the detailed explanation! Much clearer now - I guess I was confused by the semantics of install: if I install both ModuleA and ModuleB in a Configuration module, can objects injected by ModuleB be instantiated using objects injected by ModuleA? (I guess the answer is yes but I would like to make sure).

Also I hadn't noticed how ConfigurationModule is actually installing not only the parameters passed to the constructor, but also the fields defined inside of it (with reflection).

Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/ContactsUpdaterPluginModule.scala:7]: "if I install both ModuleA and ModuleB in a Configuration module, can objects injected by ModuleB be instantiated using objects injected by ModuleA" => yes
Objects are not actually injected by either module: when Guice starts, it gathers all available modules and builds the "object graph" into the injector (cf. InjectorProvider, which is our trait which calls Guice.createInjector and builds the injector). So you can inject whatever is in the object graph from the injector, it doesn't matter which module it comes from (that's not entirely true since you can define private modules which do hide their content to outer modules, but we don't use this anywhere I know of)
